flang/test/Driver/fveclib-codegen.f90

```diff
@@ -1,7 +1,8 @@
 ! test that -fveclib= is passed to the backend
 ! -target aarch64 so that ArmPL is available
-! RUN: %flang -S -Ofast -fveclib=LIBMVEC -o - %s | FileCheck %s
-! RUN: %flang -S -Ofast -fveclib=NoLibrary -o - %s | FileCheck %s --check-prefix=NOLIB
+! REQUIRES: x86-registered-target
```

Comment:
Give your comment from the summary:
>  Given that glibc only supports libmvec on aarch64 and x86_64, this test needs to target either of those to work.

shouldn't this be enabled for AArch64 as well?

Reply:
Wouldn't this become a bit involved?  One would have to require either `x86-registered-target` and run the tests with `-target x86_64-unknown-linux-gnu ` or (alternatively or both) require `aarch64-registered-target` and run them for `-target aarch64-unknown-linux-gnu`.  I thought the test was supposed to test that `libmvec` functions are used if `libmvec` is available?  Otherwise whenever `libmvec` is ported to a new target, the test would need to test that, too.

Reply:
I don't fully understand what the issue is - the tests are running on AArch64 today already, right?

Also, this LIT example might be helpful: https://github.com/llvm/llvm-project/blob/2c8bd4a729a042ed823547ea0eacad429c50d7bf/flang/test/Driver/tune-cpu-fir.f90#L5

Reply:
> I don't fully understand what the issue is - the tests are running on AArch64 today already, right?

Sure, but the test is always using the native target, irrespective if that supports `libmvec` (`aarch64` or `x86_64` only) or not.  Everywhere else, the test is guaranteed to `FAIL` as is.

> Also, this LIT example might be helpful:
> 
> https://github.com/llvm/llvm-project/blob/2c8bd4a729a042ed823547ea0eacad429c50d7bf/flang/test/Driver/tune-cpu-fir.f90#L5

I'd meanwhile found `flang/test/Lower/tune-cpu-llvm.f90` which is simliar.  This is the first time I learned about this`'%if` feature **without** the need for `REQUIRES: *-registered-target`.